### ISSUE TITLE
If order_with_respect_to is a foreign key of a foreign key it generates aditional queries.

### ISSUE THREAD

**Issue by rafammpp** [django-ordered-model/django-ordered-model#293]
# Example
```python
class Site(models.Model):
    ...

class Category(models.Model):
    site = models.ForeignKey(Site)

class Article(OrderedModel):
    order_with_respect_to = 'category__site'
    category = models.ForeignKey(Category)
```

If I do a queryset of Articles, it make an aditional query for every article, even if I use `Article.objects.select_related('category__site')`.

@ibaguio manage to fix this problem in c4630dca2bdafd6bac1bd2328081f524953872cf for a simple foreign key, but if you order with respect to a foreign key of a foreign key it makes a query to retrieve the category.

If I understand it well, you save the initial value of the wrt_map to the instance at init time to check for changes later on save, and then do the proper arragments if needed. wrt_map is a dict of every item in order_with_respect_to with it's respective values. But you only need it on save. My proposal is to get the saved value from the db, and then check if it's different from the current instance to be saved. Yes, we make an aditional query, but only when we need it (on save) and can be the cheapest query if we defer all values other than the pk.

I've a view that list articles. Currently it makes over a thousand queries, and with this patch implemented, or with the previous version of this package, it gets down to eight.

I'm gonna make a pull request with my proposal, see you there.

**Comment by ibaguio** [comment]
@shuckc, i'm always agains using `select_related` as the default solution to an FK problem, coz it "downloads" the entire related table, where you only need one field. My years of django experience has taught me to always default to use annotations, and only use `select_related` when you can't help it.

For @rafammpp 's problem, i think this can be solved by annotating the values on the queryset, top of mind here's how i would implement it:

```
class OrderModelQuerySet:
    def annotate_wrt(self):
        owrt = self.model.order_with_respect_to

        if any("__" in wrt for wrt in owrt):
            return self.annotate(
                **{
                    field_path: F(f"{field_path}__id")
                    for field_path in owrt
                    if "__" in field_path
                }
            )
        
        # no dunder relation found in order_with_respect_to
        # no annotations needed      
        return self

class OrderModelManager:
    def get_queryset(self):
        # Annotate the WRT by default
        return super().get_queryset().annotate_wrt()


# Sample model
class Answer(OrderdModel):
    order_with_respect_to = ("question__asker")


a = Answer.objects.all()
a.question__asker  # this will print a.question.asker_id
```

Here we are annotating the values of the related fields by default.

Note: Code above is untested, but should give you the idea.

I can write the PR for this if you think this is a viable solution.

**Comment by shuckc** [comment]
Yes that's the idea you mentioned previously, I put it together two days ago on https://github.com/django-ordered-model/django-ordered-model/pull/295/files#diff-62376346d0f6565502ee25a8a0af0478fc35c504178a023026ca9bf462e2ab0dR100

**Comment by shuckc** [comment]
You have a good point about needing a better fallback for `RelatedManagers` - if you save the object derived from one of those, it will not be annotated. My code assumes it is new object and will not upshuffle the departing wrt set. I'll add a test for it.

During model `__init__()` the annotations are not yet available, so I read the annotations during `save()`. However at that point its too late to examine the 'prior' value of double underscore relations, as they may have been changed, and repeating the DB query into a shadow object to inspect is itself not entirely without issues.

**Comment by ZipBrandon** [comment]
Can we resurrect this?
